csc_335_restaurant/src/model/Bill.java

```diff
@@ -52,6 +61,11 @@ public double calculateBill() {
 		return foodCost + tip;
 	}
 	
+	public ArrayList<FoodData> getOrder(){
+		ArrayList<FoodData> copyOrder = new ArrayList<>(this.order);
```

Comment:
Currently, this code just returns a shallow copy of the order, and because FoodData is mutable that will cause escaping reference issue.
```suggestion
		ArrayList<FoodData> copyOrder = new ArrayList<>();
                for (FoodData fd : this.order) {
                    copyOrder.add(new FoodData(fd));  // new instance each time
                }
```